lib/veri.ml

```diff
@@ -0,0 +1,207 @@
+    {<report = report; selves = None; >}
+
+  method report = report
+  method set_description s = {<descr = Some s >}
+end
+
+let create_move_event tag cell' data' =  
+  Bap_value.create tag Move.({cell = cell'; data = data';})   
+
+let create_mem_store = create_move_event Event.memory_store
+let create_mem_load  = create_move_event Event.memory_load
+let create_reg_read  = create_move_event Event.register_read
+let create_reg_write = create_move_event Event.register_write
+
+let is_imm_var var = 
+  let open Type in
+  match Var.typ var with
+  | Imm _ -> true
+  | _ -> false
+
+let is_mem_var var = 
+  let open Type in
+  match Var.typ var with
+  | Mem _ -> true
+  | _ -> false
+
+let imm_result v = 
+  let open Bil in
+  match v with
+  | Imm x -> Some x
+  | _ -> None
+
+let mem_result v = 
+  let open Bil in
+  match v with
+  | Mem x -> Some x
+  | _ -> None
+
+type 'a e = (event option, 'a) Monad.State.t
+
+class ['a] t is_interesting = object(self)
+  constraint 'a = #context
+  inherit ['a] Veri_traci.t as super
+
+  method private update_event ev =
+    if is_interesting ev then
+      get () >>= fun s -> put @@ s#register_event ev
+    else do_nothing ()
+        
+  method! lookup var : 'a r =
+    super#lookup var >>= fun r -> 
+    if is_imm_var var then 
+      match imm_result (value r) with 
+      | Some data -> 
+        self#update_event (create_reg_read var data) >>= fun () ->
+        return r
+      | None -> return r
+    else return r
+
+  method! update var result : 'a u = 
+    super#update var result >>= fun () -> 
+    if is_imm_var var then 
+      match imm_result (value result) with 
+      | Some data -> self#update_event (create_reg_write var data)
+      | None -> do_nothing ()
+    else do_nothing ()
+
+  method private eval_mem_event tag addr data : 'a e = 
+    match value addr, value data with
+    | Bil.Imm addr, Bil.Imm data ->
+      let ev = create_move_event tag addr data in
+      return (Some ev)
+    | _ -> return None
+
+  method! eval_store: mem:exp -> addr:exp -> exp -> endian -> size -> 'a r =
+    fun ~mem ~addr data endian size ->
+      super#eval_store ~mem ~addr data endian size >>= fun r ->
+      self#eval_exp addr >>= fun addr ->
+      self#eval_exp data >>= fun data ->
+      self#eval_mem_event Event.memory_store addr data >>=
+      fun ev -> match ev with
+      | None -> return r
+      | Some ev -> self#update_event ev >>= fun () -> return r
+
+  method! eval_load: mem:exp -> addr:exp -> endian -> size -> 'a r =
+    fun ~mem ~addr endian size ->
+      super#eval_load ~mem ~addr endian size >>= fun r ->
+      self#eval_exp addr >>= fun addr ->
+      self#eval_mem_event Event.memory_load addr r >>=
+      fun ev -> match ev with
+      | None -> return r
+      | Some ev -> self#update_event ev >>= fun () -> return r
+
+  method private eval_chunk arch chunk = 
+    match Veri_helpers.bil_of_chunk arch chunk with
+    | Error _ -> do_nothing ()
```

Comment:
hmm... why `do_nothing ()`? This should be reported as unlifted/undisassembled instruction.